datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -932,6 +932,7 @@ fn string_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataType>
         (LargeUtf8, Utf8) => Some(LargeUtf8),
         (Utf8, LargeUtf8) => Some(LargeUtf8),
         (LargeUtf8, LargeUtf8) => Some(LargeUtf8),
+        (Utf8View, Utf8View) => Some(Utf8View),
```

Comment:
Should we handle `(Utf8View, Utf8)`, `(Utf8, Utf8View)` etc?
It's possible that we read a StringViewArray from parquet, and compare it against a `Utf8`. What do you think the result type should look like? @alamb

Reply:
I agree we should permit to/coercion from any "string" / "binary" type to the associated view type.

It is probably good to make sure the coercion does the cheap direction when possible

For example,  we shouldn't be converting `StringViewArray` --> `StringArray` as that will copy all the strings.  Instead we should convert `StringArray` --> `StringViewArray` which is relatively cheap (we just have to copy the views)

I think this feature -- coercion -- is probably worth its own ticket and doesn't need to be done in this PR. I'll file a follow on ticket in the morning

Reply:
I handled the `Utf8View` coercion in this PR, but it would be better to handle the `BinaryView` in a follow-up.

Reply:
The problem with `(Utf8View, Utf8)` is that it can be quite common to have the following queries:

```sql
select * from t where c1 <> 'small';
```

Where the c1 is read to be `StringViewArray`, then we need to decide what is the result type (in order to implement the coercion). I agree with @alamb that `StringViewArray` is generally cheaper than `StringArray`, so we might just use `Utf8View` if at least one operand is using `Utf8View`.

Reply:
> I handled the `Utf8View` coercion in this PR, but it would be better to handle the `BinaryView` in a follow-up.

Filed https://github.com/apache/datafusion/issues/10996 to track Binary view support

Reply:
> The problem with `(Utf8View, Utf8)` is that it can be quite common to have the following queries:
> 
> ```sql
> select * from t where c1 <> 'small';
> ```
> 
> Where the c1 is read to be `StringViewArray`, then we need to decide what is the result type (in order to implement the coercion). I agree with @alamb that `StringViewArray` is generally cheaper than `StringArray`, so we might just use `Utf8View` if at least one operand is using `Utf8View`.

I wrote a bunch of tests in https://github.com/apache/datafusion/pull/10997 and the coercion mostly seems to work well. There is one exception which i think is minor and i will file a ticket about